rx/linq/observable/catch.py

```diff
@@ -15,7 +15,7 @@ def subscribe(observer):
 
         def on_error(exception):
             try:
-                result = handler(exception)
+                result = handler(exception, source)
```

Comment:
Instead of a backwards incompatible change here consider using the [`adapt_call`](https://github.com/ReactiveX/RxPY/blob/master/rx/internal/utils.py#L14) utility function.

This will allow existing code to work with the singe argument form effectively ignoring the source. New handlers written against this feature can accept two arguments and receive both the exception and the source.

Reply:
thanks for the suggestion!

Reply:
done, with coverage.